kernel/src/engine/arrow_expression.rs

```diff
@@ -364,7 +364,18 @@ fn apply_schema(array: &dyn Array, schema: &DataType) -> DeltaResult<RecordBatch
         ));
     };
     let applied = apply_schema_to_struct(array, struct_schema)?;
-    Ok(applied.into())
+    let (fields, columns, nulls) = applied.into_parts();
+    if let Some(nulls) = nulls {
+        if nulls.null_count() != 0 {
+            return Err(Error::invalid_struct_data(
+                "Top-level nulls in struct are not supported",
+            ));
+        }
+    }
+    Ok(RecordBatch::try_new(
+        Arc::new(ArrowSchema::new(fields)),
+        columns,
+    )?)
```

Comment:
I included this change so that we could leverage the existing arrow_expression infra with the new changes. without this we will panic within arrow on some of the tests I have for top-level nulls (instead of just returning an error)

Reply:
correct!